src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointEvidence.java

```diff
@@ -0,0 +1,500 @@
+        output.writeShort(eventWidth);
+        output.writeInt(eventStartPosition);
+        kryo.writeObject(output, templateName);
+        output.writeByte(templateEnd.ordinal());
+    }
+
+    public int getContigIndex() { return contigIndex; }
+    public int getEventStartPosition() { return eventStartPosition; }
+    public int getEventWidth() { return eventWidth; }
+    public int getContigEnd() { return eventStartPosition +eventWidth; }
+    public String getTemplateName() { return templateName; }
+    public TemplateEnd getTemplateEnd() { return templateEnd; }
+
+    @Override
+    public int compareTo( final BreakpointEvidence that ) {
+        if ( this == that ) return 0;
+        int result = Short.compare(this.contigIndex, that.contigIndex);
+        if ( result == 0 ) {
+            result = Integer.compare(this.eventStartPosition, that.eventStartPosition);
+            if ( result == 0 ) {
+                result = Short.compare(this.eventWidth, that.eventWidth);
+                if ( result == 0 ) {
+                    result = this.templateEnd.compareTo(that.templateEnd);
+                    if ( result == 0 ) {
+                        result = this.templateName.compareTo(that.templateName);
+                        if ( result == 0 ) {
+                            result = this.getClass().getName().compareTo(that.getClass().getName());
+                        }
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return contigIndex + "[" + eventStartPosition + ":" + getContigEnd() + "] " + templateName + templateEnd;
+    }
+
+    @Override
+    public boolean equals( final Object obj ) {
+        return obj instanceof BreakpointEvidence && compareTo((BreakpointEvidence)obj) == 0;
+    }
+
+    @Override
+    public int hashCode() {
+        final int mult = 1103515245;
+        int result = 12345;
+        result = mult * result + contigIndex;
+        result = mult * result + eventStartPosition;
+        result = mult * result + eventWidth;
+        result = mult * result + templateName.hashCode();
+        result = mult * result + templateEnd.hashCode();
+        result = mult * result + getClass().getSimpleName().hashCode();
+        return mult * result;
+    }
+
+    private static TemplateEnd findTemplateEnd( final GATKRead read ) {
+        return !read.isPaired() ? TemplateEnd.UNPAIRED :
+               !read.isFirstOfPair() && !read.isSecondOfPair() ? TemplateEnd.PAIRED_UNKNOWN :
+                read.isFirstOfPair() && !read.isSecondOfPair() ? TemplateEnd.PAIRED_FIRST :
+               !read.isFirstOfPair() &&  read.isSecondOfPair() ? TemplateEnd.PAIRED_SECOND :
+                       TemplateEnd.PAIRED_INTERIOR;
+    }
+
+    @DefaultSerializer(SplitRead.Serializer.class)
+    public static final class SplitRead extends BreakpointEvidence {
+        private static final short UNCERTAINTY = 2;
```

Comment:
Brainstorming a thought here but it seems like we might be able to get rid of this uncertainty if we compared the cigars of the two alignments (the one in the primary read and the one in the supplementary) and computed the microhomology/microinsertion size between them. I'll create a ticket to investigate that.